pavelib/utils/db_utils.py

```diff
@@ -202,45 +216,48 @@ def create_tarfile_from_db_cache(fingerprint, files, path):
     """
     Create a tar.gz file with the current bokchoy DB cache files.
     """
-    zipfile_name = '{}.tar.gz'.format(fingerprint)
-    zipfile_path = os.path.join(path, zipfile_name)
-    with tarfile.open(name=zipfile_path, mode='w:gz') as tar_file:
+    targz_file_name = '{}.tar.gz'.format(fingerprint)
+    targz_file_path = os.path.join(path, targz_file_name)
+    with tarfile.open(name=targz_file_path, mode='w:gz') as tar_file:
         for name in files:
             tar_file.add(os.path.join(path, name), arcname=name)
-    return zipfile_name, zipfile_path
+    return targz_file_name, targz_file_path
 
 
-def upload_to_s3(file_name, file_path, bucket_name):
+def upload_to_s3(file_name, file_path, s3_bucket_name):
     """
-    Upload the specified files to an s3 bucket.
+    Upload the specified file to an s3 bucket.
     """
-    print("Uploading {} to s3 bucket {}".format(file_name, bucket_name))
+
+    with open(file_path) as f:
+        file_content = f.read()
+
+    client = boto3.client("s3")
+    s3_key = file_name
+    print("Uploading cache file {} to s3 at bucket {}, key {}".format(file_name, s3_bucket_name, s3_key))
     try:
-        conn = boto.connect_s3()
-    except boto.exception.NoAuthHandlerFound:
+        client.put_object(Bucket=s3_bucket_name, Key=s3_key, ACL='public-read', Body=file_content)
+    except botocore.exceptions.NoCredentialsError:
         print("No AWS credentials found. "
               "Continuing without uploading the new cache to S3.")
         return
-    try:
-        bucket = conn.get_bucket(bucket_name)
-    except boto.exception.S3ResponseError:
-        print("Unable to connect to cache bucket with these credentials. "
-              "Continuing without uploading the new cache to S3.")
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] == "NoSuchBucket":
+            print("S3 bucket {} does not exist".format(s3_bucket_name))
+        else:
+            print("Unable to upload cache file to S3. "
+                  "Continuing without uploading the new cache to S3. Code: %s, Message: %s" % (
+                      e.response["Error"]["Code"], e.response["Error"]["Message"]
+                  ))
         return
-    key = boto.s3.key.Key(bucket=bucket, name=file_name)
-    bytes_written = key.set_contents_from_filename(file_path, replace=False, policy='public-read')
-    if bytes_written:
-        msg = "Wrote {} bytes to {}.".format(bytes_written, key.name)
-    else:
-        msg = "File {} already existed in bucket {}.".format(key.name, bucket_name)
-    print(msg)
+    print("Cache file {} saved to S3 at bucket {}, key {}.".format(file_name, s3_bucket_name, s3_key))
 
 
 def upload_db_cache_to_s3(fingerprint, bokchoy_db_files, bucket_name):
     """
     Update the S3 bucket with the bokchoy DB cache files.
     """
-    zipfile_name, zipfile_path = create_tarfile_from_db_cache(
```

Comment:
@alanoe, this is not needed right? A `tar.gz` file is still a `zipped` file since that it is what `gzip` does. Let's try to get rid of unnecessary changes from an already large PR.

Reply:
Although both zip and gzip use the deflate compression algorithm, zip compresses several files individually and gzip compress a single archive file, eg tar, so both operate differently; I was just trying to make the code clearer. Anyway, I agree this does not belong in this patch and I'll remove it.